cmd/icingadb/main.go

```diff
@@ -67,8 +72,38 @@ func run() int {
 		}
 	}
 
-	if err := icingadb.CheckSchema(context.Background(), db); err != nil {
-		logger.Fatalf("%+v", err)
+	{
+		err := retry.WithBackoff(
+			context.Background(),
+			func(ctx context.Context) (err error) {
+				return database.AutoUpgradeSchema(
```

Comment:
I would love to see some logging here. The `AutoUpgradeSchema` function just starts upgrading the schema without any notice.

From a user's point of view, Icinga DB just "blocked" and Icinga Web will hint that Icinga DB might have stopped. Thus, even when a user have consulted the logs and sees nothing, they will likely restart Icinga DB and crash the migration.